detect_secrets/plugins/ibm_cos_hmac.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import absolute_import
+
+import datetime
+import hashlib
+import hmac
+
+import requests
+
+from .base import RegexBasedDetector
+from detect_secrets.core.constants import VerifiedResult
+
+
+class IbmCosHmacDetector(RegexBasedDetector):
+    """Scans for IBM Cloud Object Storage HMAC credentials."""
+    # requires 3 factors
+    #
+    #   access_key: access_key_id
+    #   secret_key: secret_access_key
+    #   host, defaults to 's3.us.cloud-object-storage.appdomain.cloud'
+
+    secret_type = 'IBM COS HMAC Credentials'
+
+    token_prefix = r'(?:(?:ibm)?[-_]?cos[-_]?(?:hmac)?|)'
+    password_keyword = r'(?:secret[-_]?(?:access)?[-_]?key)'
+    password = r'([a-f0-9]{48})'
+    denylist = (
+        RegexBasedDetector.assign_regex_generator(
+            prefix_regex=token_prefix,
+            secret_keyword_regex=password_keyword,
+            secret_regex=password,
+        ),
+    )
+
+    def verify(self, token, content):
+        key_id_matches = find_access_key_id(content)
+
+        if not key_id_matches:
+            return VerifiedResult.UNVERIFIED
+
+        try:
+            for key_id in key_id_matches:
+                verify_result = verify_ibm_cos_hmac_credentials(
+                    key_id, token,
+                )
+                if verify_result is True:
```

Comment:
Probably don't need the `is True` portion of this.

Reply:
Comment addressed in latest commit.

Reply:
Comment addressed in latest commit.